### PR TITLE
New dues table, entry, views, reports, 2 migrations one imports from …

### PR DESCRIPTION
**Pull request by jrtaylor-com**
…old dues

Run table migration before import

Tests needed:
Only authorized users should see Mundane name on Dues Reports
Create/Read/Revoke
Links

### CODE REVIEW

File: orkui/controller/controller.Admin.php [hunk 1]
```diff
@@ -839,6 +839,29 @@ public function player($id) {
...
+								'MundaneId' => $id,
+								'ParkId' => valid_id($this->request->Admin_player->ParkId)?$this->request->Admin_player->ParkId:0,
+								'KingdomId' => valid_id($this->request->Admin_player->KingdomId)?$this->request->Admin_player->KingdomId:0,
+								'DuesFrom' => $this->request->Admin_player->DuesFrom,
+								'Terms' => $this->request->Admin_player->Terms,
+								'DuesForLife' => $this->request->Admin_player->DuesForLife
+							));
+						break;
+					case 'revokedues':
+						echo('<br/><br/>Revoke Dues called');
```
Comment (jrtaylor-com) [inline orkui/controller/controller.Admin.php:859]: Need to remove echo

File: orkui/controller/controller.Reports.php [hunk 2]
```diff
@@ -233,12 +233,21 @@ public function unwaivered($type=null) {
...
+	/* Old broken dues functionality */
 	public function duespaid($type=null) {
 		$this->template = 'Reports_roster.tpl';
 		$this->data['show_duespaid'] = 1;
 		$this->data['roster'] = $this->Reports->player_roster($type, $this->request->id, null, 1, 0, 2);
 	}
 
+	/* New Cooler Dues functionality */
+	public function dues($type=null) {
+		// TODO: totally dupe and change up this template
```
Comment (jrtaylor-com) [inline orkui/controller/controller.Reports.php:245]: Remove todo as it is done

**Comment by kenwalker** [comment]
The existing method of setting and retrieving dues paid no longer works. When retrieving player information, "DuesThrough"  is empty. If you set it via the existing player via UpdatePlayer with "DuesDate" and "DuesSemesters" it does not propagate thru to the new table and thus does not show up in your dues list.
The player page still has the old entry area for dues and could be punted.

**Comment by jrtaylor-com** [comment]
> The existing method of setting and retrieving dues paid no longer works. When retrieving player information, "DuesThrough" is empty. If you set it via the existing player via UpdatePlayer with "DuesDate" and "DuesSemesters" it does not propagate thru to the new table and thus does not show up in your dues list.
> The player page still has the old entry area for dues and could be punted.

The old dues bit will be greyed out(disabled), and I will add a popup when they try to change it. I am leaving some of the old UI accessible for a short while during the transition phase as people get up to speed. Thanks for the reminder to change that UI

**Comment by kenwalker** [comment]
> The old dues bit will be greyed out(disabled), and I will add a popup when they try to change it. I am leaving some of the old UI accessible for a short while during the transition phase as people get up to speed. Thanks for the reminder to change that UI

Ok that's cool. But the existing API, that can be made to work, correct?

File: system/lib/ork3/class.Player.php [hunk 3]
```diff
@@ -1220,6 +1223,96 @@ public function RemoveAward($request) {
...
+		$dues->clear();
+
+		if (Ork3::$Lib->authorization->HasAuthority($mundane_id, AUTH_PARK, $request['ParkId'], AUTH_EDIT)) {
+			$dues->mundane_id = $request['MundaneId'];
+			$dues->created_by = Ork3::$Lib->authorization->IsAuthorized($request['Token']);
+			$dues->created_on = date('Y-m-d');
+			$dues->park_id = $request['ParkId'];
+			$dues->kingdom_id = $request['KingdomId'];
+			$dues->dues_from = date('Y-m-d', strtotime($request['DuesFrom']));
+			// TODO: create private function that determins DuesUntil based on kingdom configured terms
```
Comment (jrtaylor-com) [inline system/lib/ork3/class.Player.php:1239]: Remove TODO as it is done

File: system/lib/ork3/class.Player.php [hunk 4]
```diff
@@ -1220,6 +1223,96 @@ public function RemoveAward($request) {
...
+        if (valid_id($request['MundaneId'])) {
+            $this->dues->clear();
+            $this->dues->mundane_id = $request['MundaneId'];
+			if (!empty($request['ExcludeRevoked'])) {
+				$this->dues->revoked = 0;
+			}
+			if (!empty($request['Active'])) {
+				$this->dues->dues_until_conjunction = ' AND ( `dues_for_life` = 1 OR ';
+				$this->dues->dues_until_term = "> '" . date('Y-m-d') . "') " . ' AND "" = ' ;
+				//$this->dues->dues_until_term_with = ' herehowaboutthis ' ;
```
Comment (jrtaylor-com) [inline system/lib/ork3/class.Player.php:1269]: Remove unneeded comment

File: system/lib/ork3/class.Player.php [hunk 5]
```diff
@@ -1220,6 +1223,96 @@ public function RemoveAward($request) {
...
+                        'DuesFrom' => $this->dues->dues_from,
+                        'DuesForLife' => $this->dues->dues_for_life,
+                        'Revoked' => $this->dues->revoked
+                    );
+            } while ($this->dues->next());
+        }
+        return $dues;
+	}
+
+	// TODO:
```
Comment (jrtaylor-com) [inline system/lib/ork3/class.Player.php:1293]: Remove TODO as it is done

**Comment by jrtaylor-com** [comment]
> > The old dues bit will be greyed out(disabled), and I will add a popup when they try to change it. I am leaving some of the old UI accessible for a short while during the transition phase as people get up to speed. Thanks for the reminder to change that UI
> 
> Ok that's cool. But the existing API, that can be made to work, correct?

I'm sure, point me to how you are accessing the methods and I can update the API to use what was just added

**Comment by kenwalker** [comment]
For setting Dues should be this line in UpdatePlayer:
https://github.com/amtgard/ORK3/blob/master/system/lib/ork3/class.Player.php#L810

For retrieving Dues in the Player information should be here:
https://github.com/amtgard/ORK3/blob/master/system/lib/ork3/class.Player.php#L216

**Comment by kenwalker** [comment]
Confirmed player with no privs cannot see mundane names
Confirmed player with park privs can only see park mundane names
Confirmed player with kingdom can see kingdom/park mundane names
Confirmed player with added auth capabilities can only see mundane names from that auth

**Comment by kenwalker** [comment]
If you do not have the privilege to revoke dues but try you get redirected to a web page that just says "Revoke dues called"

**Review by esdraelon — APPROVED** [review]
excllent flavor. a large serving, but delectable.

**Comment by jrtaylor-com** [comment]
@kenwalker pushed a new commit to fix the revoke no auth blank page issue, disabled the old dues UI, added Dues to the get_player method

**Comment by kenwalker** [comment]
This shouldn't have been merged yet. I haven't approved it and the backwards compatibility with existing APIs.

**Pull request merged**